### PR TITLE
Added endpoint to download original uploaded PDF file. Can implement …

### PR DESCRIPTION
**Pull request by ccjoel**
…displaying

original PDF using this endpoint.

### CODE REVIEW

File: api/requirements.txt [hunk 1]
```diff
@@ -31,6 +31,7 @@ requests==2.25.1
 python-dotenv==0.17.1
 PyYAML==5.4.1
 aioredis>=1.3.1,<1.4.0
+aiofiles==22.1.0
```
Comment (ccjoel) [inline api/requirements.txt]: Library needed in order to return a `FileResponse`.

File: api/src/documents.py [hunk 2]
```diff
@@ -522,3 +523,31 @@ def upload_file(
         },
         content=json.dumps({"id": document_id}|final_document),
     )
+
+
+@router.get("/documents/{document_id}/file")
```
Comment (ccjoel) [inline api/src/documents.py:528]: So that users can retrieve the document on the S3 location.

File: api/src/documents.py [hunk 3]
```diff
@@ -522,3 +523,31 @@ def upload_file(
...
+        logger.error(e)
+        return Response(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            headers={"msg": f"Error: {e}"},
+            content=json.dumps({"error": "Document has no uploaded source file."})
+        )
+
+    file = get_rawfile(path=s3_url)
+    headers = {'Content-Disposition': f'attachment; filename="{file_name}"'}
+    file_obj = file.detach()
```
Comment (ccjoel) [inline api/src/documents.py]: Is there a better way to do this? I wasn't sure if I could return back a buffer as the response, so had to download the file from S3 to a tmp location, then return the obj location on response.

File: api/src/documents.py [hunk 4]
```diff
@@ -522,3 +523,31 @@ def upload_file(
...
+    location = f"tmp/{file_name}"
+    with open(location, "wb") as new_file:
+        new_file.write(file_obj.read())
+
+    file_obj.close()
+
+    return FileResponse(location, media_type="application/octet-stream", filename=file_name)
```
Comment (mattprintz) [inline api/src/documents.py]: Saving a temp file is probably not needed and could cause problems. Can't we just return a normal response with the headers and media type you already have there?

Comment (ccjoel) [reply api/src/documents.py]: I wasn't able to, somehow. Do we do this anywhere else in the project? If not, Maybe we can quickly look at this together

**Comment by ccjoel** [comment]
Merging this, as we went through and addressed feedback while in a huddle.

File: api/src/documents.py (hunk 4 above)
Comment (ccjoel) [reply api/src/documents.py]: Addressed.

**Pull request merged**